azurerm/resource_arm_netapp_pool.go

```diff
@@ -0,0 +1,220 @@
+		}
+	}
+
+	location := azure.NormalizeLocation(d.Get("location").(string))
+	serviceLevel := d.Get("service_level").(string)
+	size := int64(d.Get("size").(int))
+
+	capacityPoolParameters := netapp.CapacityPool{
+		Location: utils.String(location),
+		PoolProperties: &netapp.PoolProperties{
+			ServiceLevel: netapp.ServiceLevel(serviceLevel),
+			Size:         utils.Int64(size),
+		},
+	}
+
+	future, err := client.CreateOrUpdate(ctx, capacityPoolParameters, resourceGroup, accountName, name)
+	if err != nil {
+		return fmt.Errorf("Error creating NetApp Pool %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for creation of NetApp Pool %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, accountName, name)
+	if err != nil {
+		return fmt.Errorf("Error retrieving NetApp Pool %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+	if resp.ID == nil {
+		return fmt.Errorf("Cannot read NetApp Pool %q (Resource Group %q) ID", name, resourceGroup)
+	}
+	d.SetId(*resp.ID)
+
+	return resourceArmNetAppPoolRead(d, meta)
+}
+
+func resourceArmNetAppPoolRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).Netapp.PoolClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	accountName := id.Path["netAppAccounts"]
+	name := id.Path["capacityPools"]
+
+	resp, err := client.Get(ctx, resourceGroup, accountName, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] NetApp Pools %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading NetApp Pools %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.Set("name", name)
+	d.Set("resource_group_name", resourceGroup)
+	d.Set("account_name", accountName)
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+	if poolProperties := resp.PoolProperties; poolProperties != nil {
+		if err := d.Set("service_level", poolProperties.ServiceLevel); err != nil {
+			return fmt.Errorf("Error setting `service_level`: %+v", err)
+		}
+
+		if err := d.Set("size", poolProperties.Size); err != nil {
+			return fmt.Errorf("Error setting `size`: %+v", err)
+		}
+	}
+
+	return nil
+}
+
+func resourceArmNetAppPoolDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).Netapp.PoolClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	accountName := id.Path["netAppAccounts"]
+	name := id.Path["capacityPools"]
+
+	_, err = client.Delete(ctx, resourceGroup, accountName, name)
```

Comment:
I saw that this returns a future, is there a reason we're using `waitForNetAppPoolToBeDeleted` instead? I've noticed that futures aren't always accurate so I just wanted to confirm that that was the case here

Reply:
Yes. The future isn't accurate here. If I don't use "waitForNetAppPoolToBeDeleted" here, it would throw below error.
Error Message:
azurerm_resource_group.example: Refreshing state... [id=/subscriptions/xxxxx/resourceGroups/neiltest21-resources]
azurerm_netapp_account.example: Refreshing state... [id=/subscriptions/xxx/resourceGroups/neiltest21-resources/providers/Microsoft.NetApp/netAppAccounts/neiltest21-netappaccount]
azurerm_netapp_account.example: Destroying... [id=/subscriptions/xxxx/resourceGroups/neiltest21-resources/providers/Microsoft.NetApp/netAppAccounts/neiltest21-netappaccount]
Error: Error deleting NetApp Account "neiltest21-netappaccount" (Resource Group "neiltest21-resources"): netapp.AccountsClient#Delete: Failure sending request: StatusCode=0 -- Original Error: autorest/azure: Service returned an error. Status=<nil> Code="CannotDeleteResource" Message="Can not delete resource before nested resources are deleted."

Reply:
Perfect! That makes sense